brake/backends/cachebe.py

```diff
@@ -51,12 +52,13 @@ def _keys(self, func_name, request, ip=True, field=None, period=None):
 
     def count(self, func_name, request, ip=True, field=None, period=60):
         """Increment counters for all relevant cache_keys given a request."""
-        counters = dict((key, 1) for key in self._keys(
+        counters = dict((key, (1, time.time() + period)) for key in self._keys(
             func_name, request, ip, field, period))
         counters.update(cache.get_many(counters.keys()))
-        for key in counters:
-            counters[key] += 1
-        cache.set_many(counters, timeout=period)
+        for key, value in counters.items():
+            count, expiration = value
+            count += 1
+            cache.set(key, (count, expiration), timeout=expiration - time.time())
```

Comment:
> So when you pass a tuple into the value for cache.set how does it serialize, pickle?

I'm not sure, I just let it use the default Django cache serialization. I can check to see what the cache serializer is.

> am I missing something, or is nothing reading the expiration out of the memcached value?

What do you mean? The value is read in `cache.get_many()` in line 56 and reused to set the timeout again.